File: LeetCode/Easy/Summary Ranges/Java/summaryRanges.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> res = new ArrayList<>();
        if (nums.length == 0) return res;
        int i = 0;
        int num = 0;
        for (i=1; i<nums.length; i++) {
            if (nums[i]-nums[i-1] != 1) {
                if (i-1 == num) res.add(Integer.toString(nums[num]));
                else res.add(nums[num]+"->"+nums[i-1]);
                num = i;
            }
        }
        if (i-num == 1) res.add(Integer.toString(nums[num]));
        else res.add(nums[num] + "->" + nums[i-1]);
        return res;
    }
}

File: LeetCode/Easy/Summary Ranges/C++/summary_ranges.cc
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        string str = "";
        vector<string> res;
        if (nums.size() == 0) return res;
        int i=0, j=0;
        while(j < nums.size()-1) {
            if (((long)nums[j+1]-(long)nums[j]) == 1) {
                j++;
            } else {
                if (i != j) res.push_back(to_string(nums[i])+ "->" + to_string(nums[j]));
                else res.push_back(to_string(nums[j]));
                i=j+1;
                j++;
            }
        }
        if (i< j) res.push_back(to_string(nums[i]) + "->" + to_string(nums[j]));
        else if (j == nums.size()-1) res.push_back(to_string(nums[j]));
        return res;
    }
};
